File: 6.00.1/weeks/2/JS/convertIntToBinary.js
let num = 19;
let isNeg = false;
let result = "";

if (num < 0) {
  isNeg = true;
  num = Math.abs(num);
}

if (num === 0) {
  result = "0";
}
while (num > 0) {
  result += (num % 2).toString();
  num = Math.floor(num / 2);
}
if (isNeg) {
  result = `- ${result}`;
}

console.log(`${result} is ${num} in Binary`);

File: CS_242/Chapters/04/convert_int_to_binary.py
ogNum = -8
num = -8

if num < 0:
    isNeg = True
    num = abs(num)
else:
    isNeg = False

result = ''
if num == 0:
    result = '0'
while num > 0:
    result = str(num % 2) + result
    print(result)
    num = num//2
    print(num)
if isNeg:
    result = '-' + result

print(f"binary of {ogNum} is {result}")
